tfe/data_source_state_outputs.go

```diff
@@ -0,0 +1,260 @@
+				},
+			},
+		}, nil
+	}
+	return &tfprotov5.ReadDataSourceResponse{
+		State: &state,
+	}, nil
+}
+
+func (d dataSourceStateOutputs) ValidateDataSourceConfig(ctx context.Context, req *tfprotov5.ValidateDataSourceConfigRequest) (*tfprotov5.ValidateDataSourceConfigResponse, error) {
+	return &tfprotov5.ValidateDataSourceConfigResponse{}, nil
+}
+
+func (d dataSourceStateOutputs) readConfigValues(req *tfprotov5.ReadDataSourceRequest) (string, string, error) {
+	var orgName string
+	var wsName string
+	var err error
+
+	config := req.Config
+	val, err := config.Unmarshal(tftypes.Object{
+		AttributeTypes: map[string]tftypes.Type{
+			"workspace":    tftypes.String,
+			"organization": tftypes.String,
+			"values":       tftypes.String,
+			"id":           tftypes.String,
+		}})
+	if err != nil {
+		return orgName, wsName, fmt.Errorf("Error unmarshalling config: %v", err)
+	}
+
+	var valMap map[string]tftypes.Value
+	err = val.As(&valMap)
+	if err != nil {
+		return orgName, wsName, fmt.Errorf("Error assigning configuration attributes to map: %v", err)
+	}
+
+	err = valMap["organization"].As(&orgName)
+	if err != nil {
+		return orgName, wsName, fmt.Errorf("Error assigning 'organization' value to string: %v", err)
+	}
+	err = valMap["workspace"].As(&wsName)
+	if err != nil {
+		return orgName, wsName, fmt.Errorf("Error assigning 'workspace' value to string: %v", err)
+	}
+
+	return orgName, wsName, nil
+}
+
+type rootModule struct {
+	RootOutputs map[string]rawOutput `json:"outputs"`
+}
+
+type rawOutput struct {
+	ValueRaw     json.RawMessage `json:"value"`
+	ValueTypeRaw json.RawMessage `json:"type"`
+}
+
+type outputData struct {
+	Value cty.Value
+}
+
+type stateData struct {
+	outputs map[string]*outputData
+}
+
+func (d dataSourceStateOutputs) readStateOutput(ctx context.Context, tfeClient *tfe.Client, orgName, wsName string) (*stateData, error) {
+	log.Printf("[DEBUG] Reading the Workspace %s in Organization %s", wsName, orgName)
+	ws, err := tfeClient.Workspaces.Read(ctx, orgName, wsName)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading workspace: %v", err)
+	}
+
+	log.Printf("[DEBUG] Reading the current StateVersion for Workspace ID %s.", ws.ID)
+	sv, err := tfeClient.StateVersions.Current(ctx, ws.ID)
+	if err != nil {
+		if err == tfe.ErrResourceNotFound {
+			return nil, fmt.Errorf("Current remote state for workspace '%s' not found.", wsName)
+		}
+		return nil, fmt.Errorf("Could not read the current state for workspace '%s' : %v", wsName, err)
+	}
+
+	log.Printf("[DEBUG] Downloading State Version")
+	rawState, err := tfeClient.StateVersions.Download(ctx, sv.DownloadURL)
+	if err != nil {
+		return nil, fmt.Errorf("Error downloading state: %v", err)
+	}
+
+	log.Printf("[DEBUG] Unmarshalling state output")
+	read := bytes.NewReader(rawState)
+	src, err := ioutil.ReadAll(read)
+	if err != nil {
+		return nil, fmt.Errorf("Could not read state data: %v", err)
+	}
+	rrs := &rootModule{}
```

Comment:
All the unmarshalling for the rest of this function will eventually be put in the go-client as part of the `StateVersions` interface.